Classes/ConfigurationModifier.php

```diff
@@ -43,6 +43,12 @@ protected static function findDynamicPages(array $dynamicPagesConfiguration): ar
             self::$cache[$withPluginsCacheKey] = self::$cache[$withPluginsCacheKey] ?? self::findPagesWithPlugins($withPlugins);
             array_push($pageUids, ...self::$cache[$withPluginsCacheKey]);
         }
+        if (isset($dynamicPagesConfiguration['withDoktype'])) {
+            $withDoktype = is_array($dynamicPagesConfiguration['withDoktype']) ? $dynamicPagesConfiguration['withDoktype'] : [$dynamicPagesConfiguration['withDoktype']];
```

Comment:
Please rename the configuration option to `withDoktypes` (plural).

The doktypes should be integers, but there's no sanitaziation in place. You pass it to the QueryBuilder as `PARAM_INT_ARRAY` - meaning it's not escaped. I'm not sure this is safe.

I'd prefer to have sanitization like `$withDoktypes = array_map('intval', $withDoktypes);`.

Reply:
Done